docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,474 @@
+# Writing a simple publisher and subscriber (Rust)
+* Goal: Create and run a publisher and subscriber node using Rust.
+* Tutorial level: Beginner
+* Time: 20 minutes
+<details><summary>Background</summary>
+
+In this tutorial you will create a pair of 
+[nodes](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Nodes/Understanding-ROS2-Nodes.html) that pass information to each other via a 
+[topic](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Topics/Understanding-ROS2-Topics.html) in the form of string messages. The example used here is a simple "talker" and "listener" system; one node publishes data and the other subscribes to the topic to receive that data.
+
+Since Rust doesn't have inheritance, it's not possible to inherit from `Node` as is common practice in [`rclcpp`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) or [`rclpy`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html).
+
+The code used in these examples can be found [here](https://gitlab.com/ros21923912/simple_ros2_node/-/tree/more_simple_nodes?ref_type=heads)  
```

Comment:
Can you explain in detail what are the issues you're having building `ros2_rust` locally? I really would want the examples to be in a folder in this PR (named `examples`)

Reply:
The main problem is that clippy just won't start on my computer because it doesn't understand how the ros2 installation works with rust.